frontend/src/router/index.js

```diff
@@ -78,17 +81,27 @@ const router = createRouter({
       component: UserPage,
       props: { userRole: ROLES.REPRESENTATIVE },
       meta: { isPublic: false },
-      beforeEnter: async () => {
-        return await isAllowedToAccess(ROLES.REPRESENTATIVE)
+      beforeEnter: async (to, from, next) => {
+        const isAllowed = await isAllowedToAccess(ROLES.REPRESENTATIVE)
+        if (isAllowed) {
+          next() // Allow access
+        } else {
+          next('/error-page') // Redirect to the home page or another route
+        }
       },
     },
     {
       path: '/governance',
       name: 'governance',
       component: GovernancePage,
       meta: { isPublic: false, middleware: [hasUser] },
-      beforeEnter: async () => {
-        return await isAllowedToAccess(ROLES.REPRESENTATIVE) || await isAllowedToAccess(ROLES.ADMINISTRATOR)
+      beforeEnter: async (to, from, next) => {
+        const isAllowed = await isAllowedToAccess([ROLES.REPRESENTATIVE, ROLES.ADMINISTRATOR])
+        if (isAllowed) {
+          next() // Allow access
+        } else {
+          next('/error-page') // Redirect to the home page or another route
```

Comment:
cant this duplications be extracted into a helper method? If to me this seems like the same logic: IF not allowed, THEN redirect to error page. Which seems like something you want to be done whenever a page is not accessible to a user